perma_web/static/js/helpers/handlebars_helpers.js

```diff
@@ -14,3 +17,29 @@ Handlebars.registerHelper ('truncatechars', function (str, len) {
     }
     return str;
 });
+
+/*
+Using handlebar's compile method
+to generate templates on the fly
+*/
+HandlebarsHelpers.init = function(){
+  HandlebarsHelpers.renderAllTemplates();
+}
+
+HandlebarsHelpers.renderAllTemplates = function(){
+  // compile handlebars templates
+  $("script[type='text/x-handlebars-template']").each(function(){
+      var $this = $(this),
+          name = $this.attr('id').replace('-template', '').replace(/-/g, '_');
+      templates[name] = Handlebars.compile($this.html());
+  });
+}
+
+HandlebarsHelpers.renderTemplate = function(templateId, args) {
+  var $this = $(templateId);
+  var name = templateId.replace('-template', '').replace(/-/g, '_');
+  templates[name] = Handlebars.compile($this.html());
+  return templates[name](args)
+}
+
```

Comment:
Since you're refactoring anyway ... :)

- Why don't we take it all the way and make `renderTemplate` the only way that templates are rendered? Then we don't need the `init` function or `renderAllTemplates`, and we don't have confusing code hanging around in the remaining places that refer to `templates` directly.
- Maybe `HandlebarsHelpers.templateCache` would be clearer than `window.templates` for that variable?
- Once we get rid of direct access to `templates`, there's no need for the `replace` stuff for `templateId`.
- As an optimization, can skip the `compile` if it's already cached.
- Maybe HandlebarsHelpers should have a sweet theme song? https://soundcloud.com/hamburgerhelper/hamburger-helper-prod-retro-spectro